classes/Commands/CheckNewVersionCommand.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ */
+
+namespace PrestaShop\Module\AutoUpgrade\Commands;
+
+use Exception;
+use PrestaShop\Module\AutoUpgrade\Exceptions\DistributionApiException;
+use PrestaShop\Module\AutoUpgrade\Exceptions\UpgradeException;
+use PrestaShop\Module\AutoUpgrade\Exceptions\ZipActionException;
+use PrestaShop\Module\AutoUpgrade\Parameters\UpgradeConfiguration;
+use PrestaShop\Module\AutoUpgrade\Services\LocalVersionFilesService;
+use PrestaShop\Module\AutoUpgrade\Task\ExitCode;
+use PrestaShop\Module\AutoUpgrade\UpgradeContainer;
+use PrestaShop\Module\AutoUpgrade\VersionUtils;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class CheckNewVersionCommand extends AbstractCommand
+{
+    /** @var string */
+    protected static $defaultName = 'update:check-new-version';
+
+    protected function configure(): void
+    {
+        $this
+            ->setDescription('List Prestashop updates available for the store.')
+            ->setHelp('This command allows you to list Prestashop updates available for the store.')
+            ->addArgument('admin-dir', InputArgument::REQUIRED, 'The admin directory name.');
+    }
+
+    /**
+     * @throws DistributionApiException
+     * @throws UpgradeException
+     * @throws ZipActionException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): ?int
+    {
+        try {
+            $this->setupContainer($input, $output);
+            $this->upgradeContainer->initPrestaShopAutoloader();
+            $this->upgradeContainer->initPrestaShopCore();
+
+            $this->upgradeContainer->getState()->initDefault(
```

Comment:
we going to create a state file only to get current bersion but you already pass it to this method init with `$this->upgradeContainer->getProperty(UpgradeContainer::PS_VERSION)` I think we can remove this part and pass directly $this->upgradeContainer->getProperty(UpgradeContainer::PS_VERSION)

Reply:
this part was not necessary for the process, I removed it